tests/python/unittest/test_node_reflection.py

```diff
@@ -160,14 +162,35 @@ def test_dict():
     assert set(dir(x.__class__)) <= set(dir(x))
 
 
+def test_ndarray():
+    dev = tvm.cpu(0)
+    tvm_arr = tvm.nd.array(np.random.rand(4), device=dev)
+    tvm_arr2 = tvm.ir.load_json(tvm.ir.save_json(tvm_arr))
+    tvm.ir.assert_structural_equal(tvm_arr, tvm_arr2)
+    np.testing.assert_array_equal(tvm_arr.numpy(), tvm_arr2.numpy())
+
+
+def test_ndarray_dict():
+    dev = tvm.cpu(0)
+    m1 = {
+        "key1": tvm.nd.array(np.random.rand(4), device=dev),
+        "key2": tvm.nd.array(np.random.rand(4), device=dev),
+    }
+    m2 = tvm.ir.load_json(tvm.ir.save_json(m1))
+    tvm.ir.assert_structural_equal(m1, m2)
+
+
+def test_alloc_const():
+    dev = tvm.cpu(0)
+    dtype = "float32"
+    shape = (16,)
+    buf = tvm.tir.decl_buffer(shape, dtype)
+    data = tvm.nd.array(np.random.rand(*shape).astype(dtype), device=dev)
+    body = tvm.tir.Evaluate(0)
+    stmt = tvm.tir.AllocateConst(buf.data, dtype, shape, data, body)
+    stmt2 = tvm.ir.load_json(tvm.ir.save_json(stmt))
```

Comment:
could you double-check that you're actually exporting an NDArray into JSON here? i think this might fix it , but it would be great if we could prove it through e.g. non-empty b64ndarray key

Reply:
My read is that it does not go through `b64ndarray` mechanism, but instead goes through `repr_bytes` mechanism as @vinx13 commented. The old  `b64ndarray` mechanism should still continue to work. At some time pt perhaps we can move to  `repr_bytes`  mechanism for all cases while keeping  `b64ndarray`  for one cycle

Reply:
I've added a check of `alloc_const.data` against ref data

Reply:
thanks @vinx13! did we ever figure out which mechanism is used to serialize the ndarray though?

Reply:
It will enter this path https://github.com/apache/tvm/blob/main/src/node/serialization.cc#L156, and call the registered `set_repr_bytes`